server/src/main/java/keywhiz/service/permissions/PermissionCheckModule.java

```diff
@@ -8,13 +8,31 @@
 public class PermissionCheckModule extends AbstractModule {
 
   @Override
-  protected void configure() {}
+  protected void configure() {
+    bindToDefaultConstructor(AllowClientSecretCreationPermissionCheck.class);
+  }
 
   @Provides
   public PermissionCheck createPermissionCheck(MetricRegistry metricRegistry,
       AutomationClientPermissionCheck automationClientCheck,
-      OwnershipPermissionCheck ownershipCheck) {
-    PermissionCheck anyPermissionCheck = new AnyPermissionCheck(metricRegistry, List.of(ownershipCheck, automationClientCheck));
+      OwnershipPermissionCheck ownershipCheck,
+      AllowClientSecretCreationPermissionCheck clientSecretCreationCheck) {
+
+    List<PermissionCheck> permissionChecks = List.of(
+        ownershipCheck,
+        clientSecretCreationCheck,
+        automationClientCheck
+    );
+
+    PermissionCheck anyPermissionCheck = new AnyPermissionCheck(metricRegistry, permissionChecks);
     return anyPermissionCheck;
   }
+
+  private <T> void bindToDefaultConstructor(Class<T> clazz) {
```

Comment:
Heh. I like `clazz`. Is that idiomatic? (An old coworker always used to name his buffers `buffy` and I always got a kick out of it.)